tests/test_mockgun.py

```diff
@@ -269,6 +273,112 @@ def test_find_by_sub_entity_field(self):
+        ], ["project.Project.name"])
+        self.assertIn("project.Project.name", item)
+        self.assertEqual("project1", item["project.Project.name"])
+
+
+class TestFindOrder(TestBaseWithExceptionTests):
+    """
+    Ensure that using the 'order' argument in the find method work.
+    """
+    def setUp(self):
+        self._mockgun = Mockgun("https://test.shotgunstudio.com", login="user", password="1234")
+
+        # Create two users to assign to the pipeline configurations.
+        self._user1 = self._mockgun.create("HumanUser", {"login": "user1", "firstname": "firstname1", "email": "user1@foobar.com"})
+        self._user2 = self._mockgun.create("HumanUser", {"login": "user2", "firstname": "firstname2", "email": "user2@foobar.com"})
+
+        # Create a project for nested sub entity field
+        self._project1 = self._mockgun.create("Project", {"name": "project1", "users": [self._user1]})
+        self._project2 = self._mockgun.create("Project", {"name": "project2", "users": [self._user2]})
+
+        # Create two PipelineConfiguration entities so we can test multi-entity sorting.
+        self._pipeline_configutation_1 = self._mockgun.create("PipelineConfiguration", {
+            "code": "PipelineConfiguration1", "users": [self._user2], "project": self._project1})
+        self._pipeline_configutation_2 = self._mockgun.create("PipelineConfiguration", {
+            "code": "PipelineConfiguration2", "users": [self._user1], "project": self._project2})
+
+    def test_find_order(self):
+        # Test ascending order
+        item = self._mockgun.find_one("HumanUser", [], order=[{'field_name': 'login', 'direction': 'asc'}])
+        self.assertEqual(self._user1['id'], item['id'])
+
+        # Test descending order
+        item = self._mockgun.find_one("HumanUser", [], order=[{'field_name': 'login', 'direction': 'desc'}])
+        self.assertEqual(self._user2['id'], item['id'])
+
+    def test_find_order_linked_entity_field(self):
+        """Ensure we are able to sort data using a linked entity field."""
+        # Test ascending order
+        item = self._mockgun.find_one("PipelineConfiguration", [], order=[{'field_name': 'project.Project.name', 'direction': 'asc'}])
+        self.assertEqual(self._pipeline_configutation_1['id'], item['id'])
+
+        # Test descending order
+        item = self._mockgun.find_one("PipelineConfiguration", [], order=[{'field_name': 'project.Project.name', 'direction': 'desc'}])
+        self.assertEqual(self._pipeline_configutation_2['id'], item['id'])
+
+    def test_find_order_fields_leak(self):
+        """Ensure that additional fields passed through the order argument but NOT via the field argument are not added to the resulting fields."""
+        item = self._mockgun.find_one("HumanUser", [], fields=['email'], order=[{'field_name': 'login', 'direction': 'asc'}])
+        self.assertSetEqual(set(item.keys()), set(['id', 'type', 'email']))  # note: we don't verify the order yet
```

Comment:
Unfortunately, `assertSetEqual` was introduced in Python 2.7 and we still need to support 2.6. `assertEqual` is our only friend for this.